docs/lib/auth/fragments/js/mfa.md

```diff
@@ -102,4 +102,107 @@ class App extends Component {
 }
 
 export default withAuthenticator(App, true);
-```
\ No newline at end of file
+```
+
+## Advanced use cases
+
+### Sign-in with custom auth challenges
+
+When signing in with user name and password, you will either sign in directly or be asked to pass some challenges before getting authenticated.
+
+The `user` object returned from `Auth.signIn` will contain `challengeName` and `challengeParam` if the user needs to pass those challenges. You can call corresponding functions based on those two parameters.
+
+ChallengeName:
+
+* `SMS_MFA`: The user needs to input the code received from SMS message. You can submit the code by `Auth.confirmSignIn`.
+* `SOFTWARE_TOKEN_MFA`: The user needs to input the OTP(one time password). You can submit the code by `Auth.confirmSignIn`.
+* `NEW_PASSWORD_REQUIRED`: This happens when the user account is created through the Cognito console. The user needs to input the new password and required attributes. You can submit those data by `Auth.completeNewPassword`.
+* `MFA_SETUP`: This happens when the MFA method is TOTP(the one time password) which requires the user to go through some steps to generate those passwords. You can start the setup process by `Auth.setupTOTP`.
+
+The following code is only for demonstration purpose:
+
+```javascript
+import { Auth } from 'aws-amplify';
+
+async function SignIn() {
+    try {
+        const user = await Auth.signIn(username, password);
+        if (user.challengeName === 'SMS_MFA' ||
+            user.challengeName === 'SOFTWARE_TOKEN_MFA') {
+            // You need to get the code from the UI inputs
+            // and then trigger the following function with a button click
+            const code = getCodeFromUserInput();
+            // If MFA is enabled, sign-in should be confirmed with the confirmation code
+            const loggedUser = await Auth.confirmSignIn(
+                user,   // Return object from Auth.signIn()
+                code,   // Confirmation code  
+                mfaType // MFA Type e.g. SMS_MFA, SOFTWARE_TOKEN_MFA
```

Comment:
Do we list these MFT Types anywhere to see all available types?

Reply:
We've listed it here under advanced workflows: https://github.com/aws-amplify/docs/pull/1411/files#diff-3affded1622d3bb1b45772fcf940b72eR117